### DISCUSSION TITLE
Question about site cookies

### DISCUSSION THREAD

**Discussion by zdeltab2** [gildas-lormeau/SingleFile#1539] [General]
Can SingleFile pass site cookies (including Partition Key cookies from the Cookie API) when downloading page resources, for example there are sites protected by Cloudflare that have the "cf_clearance" cookie, and when trying to download the resources without including this cookie the request fails and returns the status code "403 Forbidden", but if the cookie is passed to the request it works without problems and returns the status code "200".

**Comment by gildas-lormeau** [comment]
It's probably technically not possible to fix this kind of problem easily. I remember that Firefox and Chrome (and maybe Safari) behave slightly differently in this respect. Do you have an example URL that illustrates the problem?

**Reply by zdeltab2** [reply]
This site generates the error 403 ["https://manhwa-es.com/manga/hunterxhunter/"](https://manhwa-es.com/manga/hunterxhunter/) when trying to open the link of an image directly, but passing the referer and cookies it works without problems and the image is displayed.

**Reply by gildas-lormeau** [reply]
Thank you! I'll do a few tests, but I have a feeling it'll be complicated. 

BTW and "funnily", it seems this technique causes browser caches to bug. Once the page is saved with SingleFile, the web site is broken too (in Firefox and Chrome).
